controllers/hvpa_controller_test.go

```diff
@@ -268,293 +285,342 @@ var _ = Describe("#TestReconcile", func() {
+			Entry("UpdateMode Auto, scale up, paradoxical scaling, replicas increases, resources per replica decreases", &data{
 				setup: setup{
-					hvpa: newHvpa("hvpa-2", target.GetName(), "label-2", minChange),
-					hpaStatus: newHpaStatus(
-						2, 3, []autoscaling.HorizontalPodAutoscalerCondition{
-							{
-								Type:   autoscaling.ScalingLimited,
-								Status: v1.ConditionTrue,
-							},
-						}),
-					vpaStatus: newVpaStatus("deployment", "3G", "500m"),
-					target:    target,
-					vpaWeight: autoscalingv1alpha1.VpaWeight(40),
+					hvpa:      newHvpa("hvpa-2", target.GetName(), "label-2", minChange),
+					hpaStatus: nil,
+					vpaStatus: newVpaStatus("deployment", "2.2G", "150m"),
+					target:    newTarget("deployment", unscaledSmall, 1),
+				},
+				action: action{
+					limitScaling: limitScale,
+				},
+				expect: expect{
+					desiredReplicas: 2,
+					resourceChange:  false,
+					resources:       unscaledSmall,
+					blockedReasons:  []hvpav1alpha1.BlockingReason{},
+				},
+			}),
+			Entry("UpdateMode Auto, scaled down, no scaling because of paradoxical scaling recommendations", &data{
+				setup: setup{
+					hvpa:      newHvpa("hvpa-2", target.GetName(), "label-2", minChange),
+					hpaStatus: nil,
+					vpaStatus: newVpaStatus("deployment", "1.8G", "150m"),
+					target:    newTarget("deployment", unscaledLarge, 3),
 				},
 				action: action{
 					limitScaling: limitScale,
 				},
 				expect: expect{
-					scalingOff:      false,
 					desiredReplicas: 3,
+					resourceChange:  false,
+					blockedReasons: []hvpav1alpha1.BlockingReason{
+						hvpav1alpha1.BlockingReasonParadoxicalScaling,
+					},
+				},
+			}),
+			Entry("UpdateMode Auto, overall scale up, paradoxical scaling, but replicas decrease - should not be considered paradoxical", &data{
+				setup: setup{
+					hvpa:      newHvpa("hvpa-2", target.GetName(), "label-2", minChange),
+					hpaStatus: nil,
+					vpaStatus: newVpaStatus("deployment", "4G", "500m"),
+					target: newTarget("deployment",
+						v1.ResourceRequirements{
+							Requests: v1.ResourceList{
+								v1.ResourceCPU:    resource.MustParse("150m"),
+								v1.ResourceMemory: resource.MustParse("1.8G"),
+							},
+						}, 3),
+				},
+				expect: expect{
+					desiredReplicas: 2,
 					resourceChange:  true,
-					scaleOutLimited: true,
-					resources:       scaledByWeight40,
-					blockedReasons: []autoscalingv1alpha1.BlockingReason{
-						autoscalingv1alpha1.BlockingReasonMinChange,
+					resources: v1.ResourceRequirements{
+						Requests: v1.ResourceList{
+							"cpu":    resource.MustParse("750m"),
+							"memory": resource.MustParse("6000000k"),
+						},
 					},
+					blockedReasons: []hvpav1alpha1.BlockingReason{},
 				},
 			}),
-			Entry("UpdateMode Auto, blocked Scaling", &data{
+			Entry("UpdateMode Auto, overall scale down, paradoxical scaling, but replicas increase - should not be considered paradoxical", &data{
```

Comment:
This is the scale down counter part of the [above](https://github.com/gardener/hvpa-controller/pull/76/files#diff-beb07c813c94dc78f2e1057a5b8f96a4R374) case, right?

Reply:
yes